aiida/cmdline/commands/cmd_devel.py

```diff
@@ -22,6 +23,43 @@ def verdi_devel():
     """Commands for developers."""
 
 
+@verdi_devel.command('revive')
+@arguments.PROCESSES()
+@options.FORCE()
+@decorators.only_if_daemon_running(message='The daemon has to be running for this command to work.')
+def devel_revive(processes, force):
+    """Revive processes that seem stuck and are no longer reachable.
+
+    Warning: This method should only be used as a last resort and when you are absolutely sure that the process is no
+    longer responsive. Before you run this command, make sure you read and understand the information below.
+
+    Sometimes processes can seem to be stuck and no longer seem to change state. There is a known bug where the process
+    task can get lost, meaning that indeed no daemon worker will ever pick it up again to continue it. The first step
+    to diagnose this situation is to make sure the daemon and RabbitMQ are running properly by running `verdi status`
+    and `verdi daemon status`. If there seems to be a problem, restart RabbitMQ and run `verdi daemon restart --reset`.
+    Once both are running properly, make sure the process is no longer reachable. Call `verdi process play PID` changing
+    the `PID` for the pk of the process. If that says that the process is no longer reachable, then it is probably that
+    the process task got lost. Only after these tasks is it recommended to run this command to revive the process.
```

Comment:
I've had a go at making the docstring a bit more compact & solution-focused (+ added the explainer from the issue below).
 
```suggestion
    """Revive processes that seem stuck and are no longer reachable.

    Warning: Use only as a last resort after you've gone through the checklist below.
    
        1. Does `verdi status` indicate that both daemon and RabbitMQ are running properly?
           If not, restart the daemon with `verdi daemon restart --reset` and restart RabbitMQ.
        2. Try `verdi process play <PID>`.
           If you receive a message that the process is no longer reachable, use `verdi devel revive <PID>`.
        
        
    Details: When RabbitMQ loses the process task before the process has completed, the process is never picked up by the daemon and will remain "stuck".
    `verdi devel revive` recreates the task, which can lead to multiple instances of the task being executed and should thus be used with caution.
```

Reply:
by the way, this got me thinking:
* is there a way for us to determine whether the task already exists?
* can `verdi process play` try/except to catch situations when a process is "stuck" and simply prompt the user to revive it, if that is the case?

Reply:
Thanks a lot for the changes, that's a lot better 😄 

> is there a way for us to determine whether the task already exists before we recreate it?

Unfortunately there is not. This is a limitation even of RabbitMQ fundamental model. To check a message is in the queue, you have to consume them. But if you consume the message it is gone. Maybe technically it could be possible to consume all tasks to check for the process ID of interest, and recreate all of them. Not sure if this would have any other problems though and what the performance would be.

> can verdi process play try/except to catch situations when a process is "stuck" and simply prompt the user to revive it, if that is the case?

It could catch the `TimeoutError` and include the suggestion in the error message. The problem is though, the `TimeoutError` could be for legitimate reasons. For example, the daemon is too busy too respond because it is doing other stuff, like handling actually live processes. I would be hesitant therefore too have a blanket statement and always recommend it. I think it needs to stay a bit hidden and a last resort if we are quite sure the timeout comes from the task actually missing.

Reply:
> It could catch the `TimeoutError` and include the suggestion in the error message. The problem is though, the `TimeoutError` could be for legitimate reasons. For example, the daemon is too busy too respond because it is doing other stuff, like handling actually live processes. I would be hesitant therefore too have a blanket statement and always recommend it. I think it needs to stay a bit hidden and a last resort if we are quite sure the timeout comes from the task actually missing.

Ok. I just want to point out that the hidden command poses a much larger burden on those users who encounter this issue (they will need to write to the mailing list / search the internet; some will give up).
Perhaps a compromise could be to do the try/except and suggest trying `verdi devel revive` in the error message.

Reply:
> Ok. I just want to point out that the hidden command puts a much larger burden on those users who encounter this issue

Well, actually the hidden command _lowers_ the current burden 😉 that's the whole reason we are adding it.  But sure, I agree it doesn't completely remove it. However, I am still afraid that users will resort to using it as a magic wand (when we advertise it too readily, or once people know about it without ample warning) and use it all the time. It might cause different (potentially worse) problems.

I would say that if we are going to publicly advertise it in the error message of `verdi process play` then why even bother putting it under `verdi devel`. If we think it should be used, we might as well put it under `verdi process revive`. I am happy to do that, but then I won't stand in for the consequences if users start using it willy-nilly and start having workchains being executed multiple times and consuming a lot of resources 😅

Reply:
> Well, actually the hidden command _lowers_ the current burden 😉

Absolutely!



> I would say that if we are going to publicly advertise it in the error message of `verdi process play` then why even bother putting it under `verdi devel`. If we think it should be used, we might as well put it under `verdi process revive`.

I still like the idea of pointing to the checklist + `verdi devel revive` command in the specific error message.

In my view, if a process becomes stuck that's a bug and our goal should be to remove the need for `verdi devel revive` (so, `verdi devel` is a good place for it in my view), but as long as the problem exists we need to put ourselves into the shoes of the users and think about how we can reduce their time to solution.